src/amber/cli/helpers/process_runner.cr

```diff
@@ -0,0 +1,106 @@
+require "sentry"
+
+module Sentry
+  class ProcessRunner
+    property processes = [] of Process
+    property process_name : String
+    property files = [] of String
+    @logger : Amber::Environment::Logger
+
+    def initialize(
+                   @process_name : String,
+                   @build_command : String,
+                   @run_command : String,
+                   @build_args : Array(String) = [] of String,
+                   @run_args : Array(String) = [] of String,
+                   files = [] of String,
+                   @logger = Amber::CLI.logger)
+      @files = files
+    end
+
+    def run
+      loop do
+        scan_files
+        sleep 1
+      end
+    end
+
+    # Compiles and starts the application
+    #
+    def start_app
+      stop_all_processes
+      start_all_processes
+    end
+
+    private def scan_files
+      file_changed = false
+
+      Dir.glob(files) do |file|
+        timestamp = get_timestamp(file)
+        if FILE_TIMESTAMPS[file]? && FILE_TIMESTAMPS[file] != timestamp
+          FILE_TIMESTAMPS[file] = timestamp
+          file_changed = true
+          log "#{file.capitalize.colorize(:light_gray)}"
+        elsif FILE_TIMESTAMPS[file]?.nil?
+          FILE_TIMESTAMPS[file] = timestamp
+          file_changed = true
+          log "Watching file: #{file.capitalize.colorize(:light_gray)}"
+        end
+      end
+
+      start_app if (file_changed)
+    end
+
+    private def start_all_processes
+      log "Compiling #{project_name}..."
+      create_all_processes
+    end
+
+    private def stop_all_processes
+      log "Terminating app #{project_name}..."
+      @processes.each do |process|
+        process.kill unless process.terminated?
+      end
+      processes.clear
+    end
+
+    private def create_all_processes
+      build_app_process
+      @processes << create_watch_process
+      sleep 3
+      create_npm_process
+    end
+
+    private def build_app_process
+      log "Building project #{project_name}..."
+      Process.run(@build_command, shell: true, output: true, error: true)
+    end
+
+    private def create_watch_process
+      log "Starting #{project_name}..."
+      Process.new(@run_command, output: true, error: true)
+    end
+
+    private def create_npm_process
+      node_log "Installing dependencies..."
+      Process.new("npm install && npm run watch", output: false, error: true, shell: true)
```

Comment:
What do you think of allowing this to be configurable?  A Procfile that specifies what should be watched and ran when files change?
```
app:
  watch: ./src/**
  run: crystal run ./src/app.cr
web:
  watch: ./src/assets/**
  run: npm run build
sidekiq:
  watch ./src/jobs/**
  run: crystal ./src/sidekiq.cr
```

Reply:
Also, What if someone doesn't want to use npm?  I would prefer this not being a dependency in the amber watch command.

Reply:
@drujensen - @faustinoaq was working on a crystal/amber solution that used WebSockets, but I'm not sure the status of it. When that is done, it will be easier to pull the npm dependency.

Reply:
@marksiemers not sure what kind of solution would use WebSockets unless you're talking about hot reloading?

In either case, I don't think we should hardcode an `npm` command in the CLI.  We should be agnostic to the users choice of front-end solution.  If someone is just creating a backend api or if they want to use a different solution, we shouldn't force them to install node.

Reply:
https://github.com/oltmannsdaniel/frontender

We could still do something like that and remove npm dependency entirely.

Reply:
> was working on a crystal/amber solution that used WebSockets, but I'm not sure the status of it.

@marksiemers The WebSocket stuff is a bit different, the idea is doing client reloading using a browser script and check file changes trough an `Amber::Pipe`. Compilation stuff using command line would need NPM though. I was a bit busy with my parents during latest weeks, so I haven't done much advance on it 😅

Reply:
@drujensen I like the idea of having a profile, maybe this can just live in the `.amber.yml`, but there seems to be a lot of configurations, app, logging, and now the watcher. I would like for for us

What about for now just add a `flag`? For example:

```
amber w --front-end
```

Until we can come up with a better solution.

Reply:
@eliasjpr I prefer we take this out until we come up with a proper solution.  Adding a flag now and then possibly removing it would be jarring to users.

Also, I was surprised to see this in a PR about logging.  It's not really related to logging so I think it should be in a separate PR anyway.

Reply:
I think that would be a good idea. This isn't a Tax bill...

Reply:
_We do need to be careful about keeping PR's on point, and making the minimum amount of changes required for the stated feature_**.

For now though I wonder if we should just release npm compile as part of watch, but not document it. We can always change it later? Or remove it and create a new PR where we can discuss that.

Reply:
I just want to get this released before Christmas.
There are a lot of differences between 0.3.6 and master. It's hard to document right now because it's different depending on master or 0.3.6.